benchmarks/src/main/java/io/druid/benchmark/FilterPartitionBenchmark.java

```diff
@@ -492,68 +453,54 @@ public void readComplexOrFilterCNF(Blackhole blackhole) throws Exception
 
     StorageAdapter sa = new QueryableIndexStorageAdapter(qIndex);
     Sequence<Cursor> cursors = makeCursors(sa, Filters.convertToCNF(dimFilter3.toFilter()));
-
-    Sequence<List<String>> stringListSeq = readCursors(cursors, blackhole);
-    List<String> strings = stringListSeq.limit(1).toList().get(0);
-    for (String st : strings) {
-      blackhole.consume(st);
-    }
+    readCursors(cursors, blackhole);
   }
 
   private Sequence<Cursor> makeCursors(StorageAdapter sa, Filter filter)
   {
     return sa.makeCursors(filter, schemaInfo.getDataInterval(), VirtualColumns.EMPTY, Granularities.ALL, false, null);
   }
 
-  private Sequence<List<String>> readCursors(Sequence<Cursor> cursors, final Blackhole blackhole)
+  private void readCursors(Sequence<Cursor> cursors, Blackhole blackhole)
   {
-    return Sequences.map(
+    final Sequence<Void> voids = Sequences.map(
         cursors,
-        new Function<Cursor, List<String>>()
-        {
-          @Override
-          public List<String> apply(Cursor input)
-          {
-            List<String> strings = new ArrayList<String>();
-            List<DimensionSelector> selectors = new ArrayList<>();
-            selectors.add(
-                input.getColumnSelectorFactory().makeDimensionSelector(new DefaultDimensionSpec("dimSequential", null))
-            );
-            //selectors.add(input.makeDimensionSelector(new DefaultDimensionSpec("dimB", null)));
-            while (!input.isDone()) {
-              for (DimensionSelector selector : selectors) {
-                IndexedInts row = selector.getRow();
-                blackhole.consume(selector.lookupName(row.get(0)));
-                //strings.add(selector.lookupName(row.get(0)));
-              }
-              input.advance();
+        input -> {
+          List<DimensionSelector> selectors = new ArrayList<>();
+          selectors.add(
+              input.getColumnSelectorFactory().makeDimensionSelector(new DefaultDimensionSpec("dimSequential", null))
+          );
+          while (!input.isDone()) {
+            for (DimensionSelector selector : selectors) {
+              IndexedInts row = selector.getRow();
+              blackhole.consume(selector.lookupName(row.get(0)));
```

Comment:
Suggested to use a real aggregator to make the load more realistic

Reply:
The input is a string so aggregators cannot run on it.

Reply:
Well, I guess "cardinality" could, but that seems like a bad load to test with since it's pretty heavy.